pynautobot/core/api.py

```diff
@@ -90,6 +91,7 @@ def __init__(
         self.base_url = base_url
         self.http_session = requests.Session()
         self.http_session.verify = verify
+        self.http_session.headers.update({"User-Agent": f"python-pynautobot/{pynautobot.__version__}"})
```

Comment:
Should we instead add it to `self.headers` above and do `self.http_session.headers.update(**self.headers)` here? It looks like the only place we currently use `self.headers` is in [the graphql query](https://github.com/nautobot/pynautobot/blob/0473056b7564632d9e01ad5fdca17fe12453e1f7/pynautobot/core/graphql.py#L126).

Reply:
Can, but that's potentially more invasive; I took the path I did because I wanted something with as few changes as possible for acceptance into `ltm-1.6`. I'm happy to do as you suggest, but since `self.http_session.headers.update(**self.headers)` means that `pynautobot.core.api.Api.__init__()` would be setting an`Authorization` header, should I also remove the additions of the `Authorization` header in:
-  `pynautobot.core.query.Request.get_openapi()`
-  `pynautobot.core.query.Request.get_version()`
- `pynautobot.core.query.Request.get_status()`

?

Not strictly necessary, but it feels cleaner to me to do it that way. And `Authorization` is also being set in `pynautobot.core.query._make_call()` and being passed by most (all?) calls to `Request()`. If we add the `Authorization` header to `self.http_session.headers`, then shouldn't the calls to `Request()` be modified to no longer pass a `token` value? Arguably, `Request` could be modified to no longer /accept/ in, but that could totally go either way.

None of this cleanup is required for the functionality provided by this patch, but it feels like a question which should be answered (if not right now, then at some point). I wouldn't be surprised if I've missed something, but this is where my head's at; thoughts?

Reply:
As with most collaborative code bases, the methods that we use for different pieces of code that carry out similar functions can obviously vary based on many factors such as haste, copy pasta, inexperience or insecurity. The addition of the Authorization header to the `get_openapi`, `get_version` and `get_status` endpoints were done as a quick break-fix PR (#174) that was more "fix it ASAP" rather than a thought out design decision. You can clearly see [one of them was completely redundant](https://github.com/nautobot/pynautobot/blob/c1c2a8f2a8b8c2a5093c0b69a6d289787b160514/pynautobot/core/query.py#L223-L226)!

So, I apologize that I opened that can of worms and I relent that the simplest solution is probably the best for now.